Tasks/AzureTestPlanV0/ciEventLogger.ts

```diff
@@ -0,0 +1,33 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+
+const area: string = 'TestManagement';
+const feature: string = 'AzureTestPlanTask';
+
+function getDefaultProps() {
+    return {
+        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        releaseid: tl.getVariable('Release.ReleaseId'),
+        builduri: tl.getVariable('Build.BuildUri'),
+        buildid: tl.getVariable('Build.Buildid')
+    };
+}
+
+export function publishEvent(properties: { [key: string]: any }): void {
+    try {
+        tl.assertAgent('2.144.0');
+        publishTelemetry(area, feature, Object.assign(getDefaultProps(), properties));
+
+    } catch (err) {
+        tl.debug('Unable to publish telemetry due to lower agent version.');
+    }
+}
+
+export function publishTelemetry(area: string, feature: string, properties: { [key: string]: any }): void {
+    const data = JSON.stringify(properties);
+    tl.debug('telemetry area: ' + area + ' feature: ' + feature + ' data: ' + data);
+    tl.command('telemetry.publish', { 'area': area, 'feature': feature }, data);
```

Comment:
should we log all the telemetry in the logs? I think its not required, you can remove data

Reply:
I picked from VsTestV2, incase of telemtry.publish cmdlet failure, you have a fallback to look for telemetry in debug logs. 

We wouldn't log all the telemetry unless debug mode is enabled.